gcloud-java-core/src/main/java/com/google/gcloud/IamPolicy.java

```diff
@@ -0,0 +1,228 @@
+      bindings.get(role).remove(first);
+      bindings.get(role).removeAll(Arrays.asList(others));
+      return self();
+    }
+
+    /**
+     * Sets the policy's etag.
+     *
+     * <p>Etags are used for optimistic concurrency control as a way to help prevent simultaneous
+     * updates of a policy from overwriting each other. It is strongly suggested that systems make
+     * use of the etag in the read-modify-write cycle to perform policy updates in order to avoid
+     * race conditions. An etag is returned in the response to getIamPolicy, and systems are
+     * expected to put that etag in the request to setIamPolicy to ensure that their change will be
+     * applied to the same version of the policy.  If no etag is provided in the call to
+     * setIamPolicy, then the existing policy is overwritten blindly.
+     */
+    protected final B etag(String etag) {
+      this.etag = etag;
+      return self();
+    }
+
+    /**
+     * Sets the version of the policy. The default version is 0, meaning only the "owner", "editor",
+     * and "viewer" roles are permitted. If the version is 1, you may also use other roles.
+     */
+    protected final B version(Integer version) {
+      this.version = version;
+      return self();
+    }
+
+    @SuppressWarnings("unchecked")
+    private B self() {
+      return (B) this;
+    }
+
+    public abstract IamPolicy<R> build();
+  }
+
+  protected IamPolicy(Builder<R, ? extends Builder<R, ?>> builder) {
+    ImmutableMap.Builder<R, Set<Identity>> bindingsBuilder = ImmutableMap.builder();
+    for (Map.Entry<R, Set<Identity>> binding : builder.bindings.entrySet()) {
+      bindingsBuilder.put(binding.getKey(), ImmutableSet.copyOf(binding.getValue()));
+    }
+    this.bindings = bindingsBuilder.build();
+    this.etag = builder.etag;
+    this.version = builder.version;
+  }
+
+  /**
+   * The map of bindings that comprises the policy.
+   */
+  public Map<R, Set<Identity>> bindings() {
+    return bindings;
+  }
+
+  /**
+   * The policy's etag.
+   *
+   * <p>Etags are used for optimistic concurrency control as a way to help prevent simultaneous
+   * updates of a policy from overwriting each other. It is strongly suggested that systems make
+   * use of the etag in the read-modify-write cycle to perform policy updates in order to avoid
+   * race conditions. An etag is returned in the response to getIamPolicy, and systems are
+   * expected to put that etag in the request to setIamPolicy to ensure that their change will be
+   * applied to the same version of the policy.  If no etag is provided in the call to
+   * setIamPolicy, then the existing policy is overwritten blindly.
+   */
+  public String etag() {
+    return etag;
+  }
+
+  /**
+   * Sets the version of the policy. The default version is 0, meaning only the "owner", "editor",
+   * and "viewer" roles are permitted. If the version is 1, you may also use other roles.
+   */
+  public Integer version() {
```

Comment:
If a service specifies field options for a policy, this field can be null.  The handling of this return value depends on what we decide in #664.